app/lib/server/server.go

```diff
@@ -0,0 +1,499 @@
+package server
+
```

Comment:
Mostly renaming plus some things learnt from the `cosi`-binary. Sooner or later `cosi` should use this library, too.